File: LivrusVirtus/wxWindows/wxRuby/CadProfissoes.rb
require 'wxruby'
require 'Cadastros'
require 'TextValidatorCtrl'
require 'ConsultasSQL'
require 'resource'
require 'RotinasGlobais'

class FrmCadProfissoes < FrmCadastros
  def initialize(parent)
    super(parent, TextValidatorCtrl::FILTER_NUMERIC, true) 
    @labDescricao.set_label('Profissão:')

    evt_button(ID_FRMCADASTROS_BTNNOVO) {|event| OnBtnNovoClick(event)}
    evt_button(ID_FRMCADASTROS_BTNPESQUISAR) {|event| OnBtnPesquisarClick(event)}
    evt_button(ID_FRMCADASTROS_BTNANTERIOR) {|event| OnBtnAnteriorClick(event)}
    evt_button(ID_FRMCADASTROS_BTNPROXIMO) {|event| OnBtnProximoClick(event)}
    evt_button(ID_FRMCADASTROS_BTNSALVAR) {|event| OnBtnSalvarClick(event)}
    evt_button(ID_FRMCADASTROS_BTNEXCLUIR) {|event| OnBtnExcluirClick(event)}
    evt_button(ID_FRMCADASTROS_BTNLIMPAR) {|event| OnBtnLimparClick(event)}
    evt_button(ID_FRMCADASTROS_BTNFECHAR) {|event| OnBtnFecharClick(event)}
  end

  def ValidaDados(bTodosDados)
    if $Rotinas.ValidaCampo(
      @edtCodigo.get_value(), Resource::STR_CODNINF) then
      return false
    end
    if bTodosDados then
  	  if $Rotinas.ValidaCampo(
        @edtDescricao.get_value(), Resource::STR_PROFNINF) then
    		return false
      end
    end
    return true
  end

  def OnBtnNovoClick(event)
    self.btnNovoClick()
    self.NovoDado($ConsSQL.Profissao('N', @edtCodigo.get_value(),""))
    event.skip()
  end

  def OnBtnPesquisarClick(event)
    self.PesquisarDados($ConsSQL.Profissao('S', 
        @edtCodigo.get_value(),""), Resource::STR_PROFNENC)
    event.skip()
  end

  def OnBtnAnteriorClick(event)
    self.registroAnterior()
    event.skip()
  end

  def OnBtnProximoClick(event)
    self.registroProximo()
    event.skip()
  end

  def OnBtnSalvarClick(event)
    self.SalvarDados(self.ValidaDados(true),
      $ConsSQL.Profissao('S', @edtCodigo.get_value(),""),
      $ConsSQL.Profissao('U', @edtCodigo.get_value(), @edtDescricao.get_value()),
      $ConsSQL.Profissao('I', @edtCodigo.get_value(), 
      @edtDescricao.get_value()), true)
    event.skip()
  end

  def OnBtnExcluirClick(event)
    self.ExcluirDados(self.ValidaDados(false),
      $ConsSQL.Profissao('D', @edtCodigo.get_value(),""))
    event.skip()
  end

  def OnBtnLimparClick(event)
    self.btnLimparClick()
    event.skip()
  end

  def OnBtnFecharClick(event)
    self.get_parent().close(true)
    event.skip()
  end
end

File: LivrusVirtus/VCPP6/wxWin/LibMySQL/CadProfissoes.cpp
#include "resource.h"
#include "Cadastros.h"
#include "CadProfissoes.h"
#include "Principal.h"
#include "ConsultasSQL.h"

BEGIN_EVENT_TABLE(CfrmCadProfissoes, CfrmCadastros)
  EVT_BUTTON(ID_FRMCADASTROS_BTNNOVO, CfrmCadProfissoes::OnBtnNovo)
  EVT_BUTTON(ID_FRMCADASTROS_BTNPESQUISAR, CfrmCadProfissoes::OnBtnPesquisar)
  EVT_BUTTON(ID_FRMCADASTROS_BTNANTERIOR, CfrmCadProfissoes::OnBtnAnterior)
  EVT_BUTTON(ID_FRMCADASTROS_BTNPROXIMO, CfrmCadProfissoes::OnBtnProximo)
  EVT_BUTTON(ID_FRMCADASTROS_BTNSALVAR, CfrmCadProfissoes::OnBtnSalvar)
  EVT_BUTTON(ID_FRMCADASTROS_BTNEXCLUIR, CfrmCadProfissoes::OnBtnExcluir)
  EVT_BUTTON(ID_FRMCADASTROS_BTNLIMPAR, CfrmCadProfissoes::OnBtnLimpar)
  EVT_BUTTON(ID_FRMCADASTROS_BTNFECHAR, CfrmCadProfissoes::OnBtnFechar)
  EVT_CLOSE(CfrmCadProfissoes::OnClose)
END_EVENT_TABLE()

CfrmCadProfissoes::CfrmCadProfissoes(wxWindow *parent)
       : CfrmCadastros(parent, wxFILTER_NUMERIC, true) {
  CfrmCadProfissoes::IniciaControles();
}

void CfrmCadProfissoes::IniciaControles() {
 this->labDescricao->SetLabel(wxT("Profissao: "));
}

CfrmCadProfissoes::~CfrmCadProfissoes() { }

bool CfrmCadProfissoes::ValidaDados(bool bTodosDados)
{
if (Rotinas->ValidaCampo(edtCodigo->GetValue(), STR_CODNINF))
  	  return false;
  	if (bTodosDados) {
  	  if (Rotinas->ValidaCampo(edtDescricao->GetValue(), STR_PROFNINF))
  		return false;
  	}
  	return true;
}

void CfrmCadProfissoes::OnClose(wxCloseEvent& event) {
  this->GetParent()->Close();
  event.Skip();
}

void CfrmCadProfissoes::OnBtnNovo(wxCommandEvent& event) {
  btnNovoClick();
  NovoDado(ConsSQL->Profissao('N', edtCodigo->GetValue(),""));	
  event.Skip();
}

void CfrmCadProfissoes::OnBtnPesquisar(wxCommandEvent& event) {
  PesquisarDados(
    ConsSQL->Profissao('S', edtCodigo->GetValue(),""), STR_PROFNENC);
  event.Skip();
}

void CfrmCadProfissoes::OnBtnAnterior(wxCommandEvent& event) {
  registroAnterior();	
  event.Skip();
}

void CfrmCadProfissoes::OnBtnProximo(wxCommandEvent& event) {
  registroProximo();		
  event.Skip();
}

void CfrmCadProfissoes::OnBtnSalvar(wxCommandEvent& event) {
  SalvarDados(ValidaDados(true),
    ConsSQL->Profissao('S', edtCodigo->GetValue(),""),
    ConsSQL->Profissao('U', edtCodigo->GetValue(), edtDescricao->GetValue()),
    ConsSQL->Profissao('I', edtCodigo->GetValue(), edtDescricao->GetValue()), 
    true);
  event.Skip();
}

void CfrmCadProfissoes::OnBtnExcluir(wxCommandEvent& event) {
  ExcluirDados(ValidaDados(false),
    ConsSQL->Profissao('D', edtCodigo->GetValue(),""));
  event.Skip();
}

void CfrmCadProfissoes::OnBtnFechar(wxCommandEvent& event) {
  this->Close(true);
  event.Skip();
}

void CfrmCadProfissoes::OnBtnLimpar(wxCommandEvent& event) {
  btnLimparClick();
  event.Skip();
}

